### ISSUE TITLE
"number" input type maps to "text" input type

### ISSUE THREAD

**Issue by tbriles** [symfony/symfony#8106]
There is even a comment in src/Symfony/Bridge/Twig/Resources/views/Form/form_div_layout.html.twig stating:

"{# type="number" doesn't work with floats #}",

which isn't true.  I'd recommend setting step="any" as a default.

**Comment by barryvdh** [comment]
Still no reply after 2 years?

I would agree that the `number` type rendering as `text` feels a bit counter-intuitive..

**Comment by jrobeson** [comment]
it doesn't render as <input type="text" />. I see <input type="number" />

**Comment by dolbertz** [comment]
To clearify: Symfony 3.0.3 renders NumberType as type="text", and IntegerType as type="number"

**Comment by AxxiD** [comment]
This issue is still persistant in v3.0.6 and was originally marked to be changed in 3.0.
Also: See my comment on https://github.com/symfony/symfony/issues/10240#issuecomment-220395462

**Comment by ianfp** [comment]
You can always override the default Twig template for `NumberType` like so:

In config.yml:

```yaml
twig:
    form_theme:
        - "form/fields.html.twig"
```

In templates/form/fields.html.twig:

```twig
{%- block number_widget -%}
    {# https://github.com/symfony/symfony/issues/8106 #}
    {%- set type = type|default('number') -%}
    {{ block('form_widget_simple') }}
{%- endblock number_widget -%}
```

This seems to work, but I've only just now started testing it.

**Comment by laufhannes** [comment]
The snippet provided by @ianfp does not work for us as alternative. If browser and php language settings do not match, problems may occur. For German numbers Symfony does automatically render `value="0,3"` which is ignored by a browser with English settings (as it expects `.` as decimal point).

**Comment by xabbuh** [comment]
will be implemented in #30267

**Comment by Volmarg** [comment]
I will just leave this here as it took me like 2 min to find out how and where:
````
            ->add('Price', NumberType::class, [
                "attr" => [
                    "min"  => 0.1,
                ],
                'label' => $this->app->translator->translate('forms.MyPaymentsProductsType.labels.price'),
                'html5' => true, // this is the part that converts text to number type
            ])
````
